diff --git a/src/Makefile.am b/src/Makefile.am
index 454b644bb..c63308790 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -2312,3 +2312,19 @@ coverage:
 endif

 .PHONY: massif update-all update-ffmpeg update-map-file coverage
+
+# vchan sink module
+module_vchan_sink_la_SOURCES = modules/module-vchan-sink.c
+module_vchan_sink_la_LDFLAGS = $(MODULE_LDFLAGS) -lvchan-xen
+module_vchan_sink_la_LIBADD = $(MODULE_LIBADD)
+module_vchan_sink_la_CFLAGS = $(AM_CFLAGS) -I/usr/include/vchan-xen -DPA_MODULE_NAME=module_vchan_sink -Dfoomodulepipecustomsymdeffoo
+
+modlibexec_LTLIBRARIES = libprotocol-native.la \
+                         module-native-protocol-unix.la \
+                         module-null-sink.la \
+                         module-null-source.la \
+                         module-always-sink.la \
+                         module-always-source.la \
+                         module-vchan-sink.la
+bin_PROGRAMS = pulseaudio pacat pactl
+lib_LTLIBRARIES = libpulse.la
